core/helpers/causeDebt.ts

```diff
@@ -0,0 +1,118 @@
+import getContract, { getContractAddressByName } from '../src/contracts';
+import BigNumber from '../src/bignumber';
+import { WAD_NUMBER_OF_DIGITS } from '../src/constants/UNITS';
+import { EthereumProvider } from 'hardhat/types';
+import { ethers } from 'ethers';
+import { formatToHex } from './format';
+
+export const causeDebt = async (
+    network: string,
+    provider: EthereumProvider,
+    debtAmountDai: BigNumber,
+    mkrOnAuction: BigNumber,
+    daiOnAuction: BigNumber
+) => {
+    await overwriteQueuedDebt(network, provider, debtAmountDai);
+    await ensureQueuedDebtEqual(network, debtAmountDai);
+    await overwriteProtocolOwnDaiBalance(network, provider);
+    await ensureDaiIsZero(network);
+    await overwriteLottedMkrAmount(network, provider, mkrOnAuction);
+    await ensureLottedMkrHasValue(network, mkrOnAuction);
+    await overwriteBidDaiAmount(network, provider, daiOnAuction);
+    await ensureBidDaiHasValue(network, daiOnAuction);
+    await startDebtAuction(network);
+};
+
+const overwriteValueInSlot = async (
+    network: string,
+    contractName: string,
+    slotAddress: string,
+    provider: EthereumProvider,
+    newValue: string
+) => {
+    const contractAddress = await getContractAddressByName(network, contractName);
+    const storageToWrite = [contractAddress, slotAddress, newValue];
+    await provider.send('hardhat_setStorageAt', storageToWrite);
+};
+
+const getMappingSlotAddress = (mappingStartSlot: string, key: string) => {
```

Comment:
- [ ] That doesn't `get` in a sense of `fetch`, but rather `generate`?
- [ ] I still suggest to create something like `overwriteUintMapping` that uses both `overwriteUintValue` and this function
- [ ] `slotAddress` and `mappingStartSlot` seems to be the same thing, no? Then please use the same variable name for them
- [ ] I suggest to move those functions into separate hardhat helpers file

Reply:
> slotAddress and mappingStartSlot seems to be the same thing, no?

this then would become very poor and confusing naming.

https://docs.soliditylang.org/en/v0.8.14/internals/layout_in_storage.html